### PR TITLE
Network Monitor Improvements

### PR DESCRIPTION
**Pull request by jasonLaster**
* Create a Table HoC
* Move filter bar to the top
* clean up request details
* simplify selecting a row
* extract some components
* Add resource headers
* Add type filtering

https://www.loom.com/share/74c8738cec944b63b74a513ce5bdedcd

### CODE REVIEW

File: src/ui/components/NetworkMonitor/RequestRow.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  // Did we just pass the row boundary that contains the current time?
+  // If so, let's render this row as the "current" row and all rows
+  // after it as future rows.
+  // if (
+  //   selectedRequest?.id === row.original.id ||
+  //   (!selectedRequest && renderingRequestsInThePast && currentTime <= row.original.point.time)
+  // ) {
+  //   renderingRequestsInThePast = false;
+  //   currentRow = true;
+  // }
```
Comment (jasonLaster) [inline src/ui/components/NetworkMonitor/RequestRow.tsx:16]: We'll need to refactor this to work in the new world... I think we could probably pass in a currentRowIndex which will be stable and an easy check

**Comment by jasonLaster** [comment]
https://app.replay.io/recording/c6850e7b-a8df-48a2-986b-9862a8a62f45?point=134999718341537425369985633627414178&time=80291&hasFrames=false

File: src/ui/components/NetworkMonitor/RequestRow.tsx (hunk 1 above)
Comment (jcmorrow) [reply src/ui/components/NetworkMonitor/RequestRow.tsx:16]: I think the struggle with currentRowIndex is when things get filtered.

File: src/ui/components/NetworkMonitor/TypesDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (jcmorrow) [inline src/ui/components/NetworkMonitor/TypesDropdown.tsx:3]: Are we supposed to be adding this to every new file, or is there some particular criteria for when we do or don't add it?

File: src/ui/components/NetworkMonitor/utils.ts [hunk 3]
```diff
@@ -96,7 +115,27 @@ export const partialRequestsToCompleteSummaries = (
         triggerPoint: r.triggerPoint,
         url: request.event.requestUrl,
       };
+    })
+    .filter(row => {
+      console.log(types);
```
Comment (jcmorrow) [inline src/ui/components/NetworkMonitor/utils.ts:120]: ✂️

**Review by jcmorrow — APPROVED** [review]
I think this looks ready to go out. The only follow-ups I would do are bringing back the timeline and using more specific material icons, as we discussed.

File: src/ui/components/NetworkMonitor/index.tsx [hunk 4]
```diff
@@ -7,31 +7,53 @@ import { getCurrentTime } from "ui/reducers/timeline";
...
+              initialSize="50%"
+              minSize={selectedRequest ? "20%" : "100%"}
+              maxSize={selectedRequest ? "80%" : "100%"}
+              endPanel={selectedRequest ? <RequestDetails request={selectedRequest} /> : <div />}
+              startPanel={
+                <RequestTable
+                  table={table}
+                  data={data}
+                  currentTime={currentTime}
+                  onRowSelect={setSelectedRequest}
```
Comment (jcmorrow) [inline src/ui/components/NetworkMonitor/index.tsx:45]: Is there actually a change to the row selection here? This seems like the same to me but I couldn't quite tell.

**Pull request merged**